galasa-ui/src/components/runs/LogTab.tsx

```diff
@@ -41,15 +49,30 @@ export default function LogTab({ logs }: { logs: string }) {
     TRACE: true
   });
 
+  // Cache for search results to avoid recomputation
+  const [searchCache, setSearchCache] = useState<Map<string, MatchInfo[]>>(new Map());
+
   const logContainerRef = useRef<HTMLDivElement>(null);
+  const debounceTimeoutRef = useRef<NodeJS.Timeout | null>(null);
+  const DEBOUNCE_DELAY_MILLISECONDS = 300;
 
   const handleSearchChange = (e: any) => {
     const value = e.target?.value || '';
     setSearchTerm(value);
-    if (!value.trim()) {
-      setCurrentMatchIndex(-1);
-      setTotalMatches(0);
+
+    // Clear any existing timeout
+    if (debounceTimeoutRef.current) {
+      clearTimeout(debounceTimeoutRef.current);
     }
+
+    // Set a new timeout
+    debounceTimeoutRef.current = setTimeout(() => {
```

Comment:
Are we able to unit test this new debounce delay behaviour to see if it searches after the given delay?